diff --git a/Makefile b/Makefile
index a03a95c..fe864c3 100644
--- a/Makefile
+++ b/Makefile
@@ -29,7 +29,7 @@ stest: stest.o
 	$(CC) -o $@ stest.o $(LDFLAGS)
 
 clean:
-	rm -f dmenu stest $(OBJ) dmenu-$(VERSION).tar.gz
+	rm -f dmenu stest $(OBJ) dmenu-$(VERSION).tar.gz config.h
 
 dist: clean
 	mkdir -p dmenu-$(VERSION)
diff --git a/dmenu.c b/dmenu.c
index 65f25ce..1e37a9a 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -634,7 +634,7 @@ setup(void)
 		/* no focused window is on screen, so use pointer location instead */
 		if (mon < 0 && !area && XQueryPointer(dpy, root, &dw, &dw, &x, &y, &di, &di, &du))
 			for (i = 0; i < n; i++)
-				if (INTERSECT(x, y, 1, 1, info[i]))
+				if (INTERSECT(x, y, 1, 1, info[i]) != 0)
 					break;
 
 		x = info[i].x_org;
